### ISSUE TITLE
CharmProductionNtuple is messy

### ISSUE THREAD

**Issue by alexpearce** [alexpearce/CharmProduction#37]
I don't like the look of all the [`var_info`](https://github.com/alexpearce/CharmProduction/blob/15c846f03e89f33587b848d990c7152d4e1efe04/analysis/charmproduction/ntuples.py#L76) and related methods (which were introduced by me, sorry!). I propose to replace them with something more along the lines of this.

```python
class CharmProductionNtuple(Ntuple.Ntuple):
    # ...
    def __init__(self, polarity, year):
        # ...
        self.mass_var = containers.Variable(...)
        # More variables here...
    # ...
```

with a new [`container` class](https://github.com/alexpearce/CharmProduction/blob/395773ce20f6ab6e6a9643ce7279cc1f77653f69/analysis/charmproduction/containers.py) called `Variable` (better names welcome!)

```python
class Variable(object):
    def __init__(self, name, title, binning, unit=''):
        """Initialise a new Variable object.

        Keyword arguments:
        name -- Branch name this Variable represents
        title -- Pretty name to display on plots etc. in TLatex format
        binning -- List of bin edges including lowest and highest
                        e.g. binning=[1, 3, 4, 5, 10]
        unit -- Physical unit as a string in TLatex format (default: '')
        """
        if sorted(binning) != binning:
            # Raise an exception or whatever, user should pass ordered binning
        self.name = name
        self.title = title
        self.binning = ROOT.RooBinning(binning[0], binning[-1])
        for b in binning[1:-2]:
            self.binning.addBoundary(b)
        self.unit = unit
````

Then rather than having [very ugly code](https://github.com/alexpearce/CharmProduction/blob/15c846f03e89f33587b848d990c7152d4e1efe04/analysis/charmproduction/fitting/utilities.py#L42-L48) expanding the [current tuples of information](https://github.com/alexpearce/CharmProduction/blob/15c846f03e89f33587b848d990c7152d4e1efe04/analysis/charmproduction/ntuples.py#L76), we can just do `ntuple.mass_var.name` or whatever.

Thoughts? I'll work on this once `fitting` is merged with `analysis`, as code outside `fitting` is using the `var_info` methods too.

**Comment by chrisburr** [comment]
This looks nice. As an extension we could have a `variables` module with a file for each variable we use so we can avoid things like defining the LaTeX names repeatedly.

**Comment by alexpearce** [comment]
Closed with merging #40.
